logger/logger_test.go

```diff
@@ -81,6 +81,7 @@ func TestLogger_Log(t *testing.T) {
 	assert.Equal(t, "testing", line["line"])
 	assert.Equal(t, "info", line["level"])
 	assert.Equal(t, "test", line["app"])
+	assert.InDelta(t, time.Now().UnixNano()/int64(time.Millisecond), int64(line["timestamp"].(float64)), 5)
```

Comment:
no need to introduce a timing constraint here with `InDelta`, that seems flakey. since we don't know the exact value when `Timestamp` is set automatically, `NotEmpty` is sufficient here.

additionally, there should be a test where we supply a `Timestamp` in options - in that case we _can_ check for an exact value with `Equal`. this doesn't necessarily need to be a separate test, we can supply a timestamp in the `LogWithOptions` test below.

Reply:
NotEmpty doesn't feel like a sufficient check imo. Am using inDelta- because results are non-deterministic as to whether the current calculated timestamp in ms matches the timestamp we've associated with the line. Agreed with the options Timestamp check- think we should also do some validation to make sure what's provided there does look like a timestamp too

Reply:
I do agree that on it's own, it would not be sufficient. What I'm suggesting is that for tests where we aren't supplying a timestamp, we simply check for its existence in the payload. In the `LogWithOptions` test, we can assert on the _exact_ value of the supplied timestamp because we know it ahead of time. IMO that is sufficient coverage.

The downside here, and why I'm somewhat opposed to this approach, is that we're relying on the timing of the tests. This is saying "timestamp is exactly 5ms or less in value from `time.Now()`" which is something we should definitely avoid doing. We aren't really gaining anything there.

Reply:
have updated the code to make the NotExists check (when it's not LogWithOptions) and to make the inDelta check when it is LogWithOptions. Reason I have to use inDelta is because timestamp (in Options) is initialized as time.Now() and so the value will change based on the timing of the tests